hicexplorer/utilities.py

```diff
@@ -35,6 +35,44 @@ def readBed(pBedFile):
     return viewpoints
 
 
+def readTargetBed(pBedFile):
+    present_genes = {}
+    targetDict = {}
+    targetPosDict = {}
+    # The keys are hardcoded here. Usually, for hdf input file, read as h5py.File().keys()
+    # These keys should match the views.hdf
+    # TODO: need to find a encode these in the BED file
+    outer_matrix = 'c_adj_norm'
+    inner_matrix = 't_adj_norm'
+    present_genes[outer_matrix] = {}
+    present_genes[outer_matrix][inner_matrix] = []
+    with open(pBedFile, 'r') as file:
+        for line in file.readlines():
+            if line.startswith('#'):
+                continue
+            _line = line.strip().split('\t')
+            if len(line) == 0:
+                continue
+            try:
+                chrom, start, end = _line[:4]
+            except ValueError:
+                _line = line.strip().split()
```

Comment:
When a line has less then 3 columns, this exception is raised, and then you are trying the same again?

Reply:
No, first split was with tabs. This one with white spaces, if the input BED file is not tab seperated.